docs/element-definitions.md

```diff
@@ -234,6 +239,49 @@ Note: At the moment only the HTTP protocol is supported.
 }
 ```
 
+### Affordance (Base API Element)
+
+An affordance element provides a way to provide affordance definitions for the
+API. These definitions can describe link relations used in a Transition Element,
+along with defining unsafe and idempotent transitions.
+
+While data structures are used for defining semantics around data models,
+affordances are used for defining semantics around hypermedia state transitions.
+
+The control types used here are based on the types used in [ALPS][]. The meta
+property `id` SHOULD be used to define the machine-usable name of the
+affordance.
+
+#### Properties
+
+- `element`: affordance
+- `attributes`
+    - `controlType` (enum)
+        - safe - A hypermedia control that triggers a safe, idempotent state
+          transition (e.g. HTTP.GET or HTTP.HEAD).
+        - unsafe - A hypermedia control that triggers an unsafe, non-idempotent
+          state transition (e.g. HTTP.POST).
+        - idempotent - A hypermedia control that triggers an unsafe, idempotent
```

Comment:
So this borrows from what has already [been done](https://tools.ietf.org/html/draft-amundsen-richardson-foster-alps-01#section-2.2.12) by Mike Amundsen and others in ALPS. It appears they get around this with overloading the "idempotent" term to mean an "unsafe, idempotent" control. This would mean that you would use `safe` for `GET` and `idempotent` for `DELETE`. Conversely, `idempotent` in this scenario would not be applied to `GET`.

I'm going to reach out to Mike and others about this, because I think you bring up a good point. While what we have solves the problem you mention, it is a little confusing to say "idempotent" must also be "unsafe." It may be better to do an enum and let users decide how they describe it. 

But one note on that—it may require that `idempotent` always has either `safe` or `unsafe` with it if this is to support automatically mapping to other protocols. I'm betting that's why they went the direction they did.

I'll post what I find here and we can chat about where to go from there.